src/lambda/worker/handler.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { S3Client } from "@aws-sdk/client-s3";
+
+import { createHandler, InjectedS3Client } from "./handler";
+
+
+let handler: ReturnType<typeof createHandler>;
+let s3Client: InjectedS3Client;
+let bucketName: string;
+
+const message = {
+    repo: "test_repo",
+    cloneUrl: "https://github.com/rangle/radius-tracker.git",
+    defaultBranch: "main",
+    repoId: "test_repoId",
+    owner: "test_owner",
+};
+
+const eventBody = {
+    Message: JSON.stringify(message),
+};
+
+const workerEvent = {
+    Records: [
+        {
+            messageId: "test_id",
+            body: JSON.stringify(eventBody),
+        },
+    ],
+};
+
+describe("Worker lambda", () => {
+    jest.useFakeTimers();
+
+    beforeEach(() => {
+        jest.runAllTimers();
+        s3Client = new S3Client({});
+        bucketName = "BUCKET_NAME_" + Math.random();
+
+        handler = createHandler(
+            s3Client,
+            {
+                BUCKET_NAME: bucketName,
+            },
+        );
+    });
+
+    it("should execute successfully", async () => {
+        s3Client.send = jest.fn().mockReturnValue(true),
```

Comment:
`send` is asynchronous, so I believe the `mockResolvedValue` should be used instead:
```suggestion
        s3Client.send = jest.fn().mockResolvedValue(true),
```

This makes little difference for the code as written, but preserves the method signature.